crates/gossip/src/peer_meta_store.rs

```diff
@@ -67,6 +67,129 @@ impl K2PeerMetaStore {
         .await
     }
 
+    /// Get the number of peer behavior errors.
+    ///
+    /// These are gossip rounds that have failed for a reason that we consider to be a wrong
+    /// behavior by the peer. This might indicate malicious behavior or a protocol mismatch.
+    pub(crate) async fn peer_behavior_errors(
+        &self,
+        peer: Url,
+    ) -> K2Result<Option<u32>> {
+        self.get(peer, "gossip:peer_behavior_errors").await
+    }
+
+    /// Increment the number of peer behavior errors for the given peer.
+    pub(crate) async fn incr_peer_behavior_errors(
+        &self,
+        peer: Url,
+    ) -> K2Result<()> {
+        let value = self.peer_behavior_errors(peer.clone()).await?;
```

Comment:
Note that these operations aren't atomic. As we are only supposed to have one session in progress with a given peer at a time, this should be safe